client/app/models/contact.coffee

```diff
@@ -148,6 +148,26 @@ module.exports = class Contact extends Backbone.Model
         _.compact([pf, fn, mn, gn, sf]).join ' '
 
 
+    # We override toJSON to ensure the exported object is the a true object.
+    # Allow a options.toVCF boolean to sanitize the output before.
+    toJSON: ({toVCF} = {}) ->
+        data = super
+        data.datapoints = data.datapoints.toJSON()
+
+        return data unless toVCF
+
+        # Isolate in a try..catch block to go outside main thread and pop error
+        try
+            # We remove the BDay datapoint if it doesn't match ISO 8601
+            unless moment(data.bday, 'YYYY-MM-DD', true).isValid()
+                delete data['bday']
+                throw new Error "Malformed bday for #{data.id}"
+        catch err
+            console.error err
```

Comment:
At last, I think that model objects should not interact or call directly with browser global variables like `console`. Instead we should use Backbone's event mechanisms to dispatch events about errors in object, and have a controller listening to this kind of event. We could also store in the object a list of errors and have an `hasErrors` method.
__But as this time__ the `console.error`is ok and will do the job well :D